### DISCUSSION TITLE
vitest with playwright mock error

### DISCUSSION THREAD

**Discussion by ezenkico** [vitest-dev/vitest#4289] [Q&A]
I'm using playwright with vitest to test my sveltekit application. I'm attempting to test a post request that is sent from one of the api endpoints via axios

I have attempted the following all with the same error.

```
const spy = vi.spyOn(axios, "post");
```
```
vi.mock("axios");
```
```
vi.mocked(axios, true).post.mockResolvedValue({
  data: {}
});
```
The error I keep getting is

`TypeError: Cannot redefine property: Symbol($$jest-matchers-object)`
Has anyone ever seen this before?

**Comment by ayazemregm** [comment]
I get the same error when I try to use component testing.

TypeError: Cannot redefine property: Symbol($$jest-matchers-object)

I just set it up and run the test that is in the [guideline](https://playwright.dev/docs/test-components)

**Comment by paumur** [comment]
I had the same issue. My problem was that in the playwright-ct.config.ts file there is testDir property. You have to change the path where your tests are located.

**Reply by ValentinoFilipetto** [reply]
Yes I had the same problem! I indeed suggest to have a look at `playwright.config.ts`

**Comment by dario-piotrowicz** [comment]
I've encountered the issue, I've tried hacking around and patching the playwright node_modules files so to avoid this jest matchers object conflict.

However after doing that I encounter a new error anyways:
![Screenshot 2024-09-22 at 11 53 22](https://github.com/user-attachments/assets/b4c6bb62-983b-46e6-b402-d87cdef385df)

So I don't think vitest utilities can just be reused outside of vitest itself.

**Reply by dario-piotrowicz** [reply]
As per this [comment](https://discord.com/channels/917386801235247114/917389678326480926/1287384201934409812), to use mocks outside of vitest itself they should simply be imported from `@vitest/spy` 🙂 

it does work for me 🥳

**Comment by kmturley** [comment]
I have the same issue in a project with both:
- vitest unit tests
- playwright end-to-end tests

Inside the `./tests` directory I have:
- api.spec.ts
- index.test.ts

playwright.config.ts
```
import { defineConfig } from '@playwright/test';
export default defineConfig({
  testDir: './tests'
});
```

```
npx playwright test --debug
TypeError: Cannot redefine property: Symbol($$jest-matchers-object)
    at /my-site/node_modules/@vitest/expect/dist/index.js:589:10
```

The issue appears to be that playwright is running both files:
- api.spec.ts
- index.test.ts

If I comment out the vitest unit tests it runs fine.
A better fix was to update `playwright.config.ts` to only match `*.spec.ts` files:

```
import { defineConfig } from '@playwright/test';

export default defineConfig({
  testDir: './tests',
  testMatch: '*.spec.ts'
});
```

and vitest.config.ts also needed updating:
```
/// <reference types="vitest/config" />
import { defineConfig } from 'vite';
export default defineConfig({
  test: {
    include: ['./tests/**/*.test.ts']
  },
});
```
